cobc/parser.y

```diff
@@ -13032,13 +13033,20 @@ enable_statement:
 
 entry_statement:
   ENTRY
+  {
```

Comment:
These are actually two different statements with the second being highly unportable and not recommend for common use; I'd prefer those to be kept separate.

Reply:
The issue here is we need to have a call to `check_non_area_a` right after reading `STOP`, without any lookahead. I admit this is kind of a fragile approach, but I did not find much better (yet).
Another means might be to find a way to attach an `in_area_a` to the `STOP` token… not sure this is straightforward.

Reply:
Hm, that's still the lookahead you want to prevent, isn't it?

Reply:
Yes I meant `ENTRY` here (but `STOP` is a similar statement that needed rework to avoid the lookahead).
In the case above we still have the lookahead. As an alternative I'll try with an `entry_word` non-terminal…

Reply:
Using dedicated non-terminals seem to circumvent the lookahead issue in a "cleaner" way. I'll push some changes in that direction.